lib/http_accept_language/auto_locale.rb

```diff
@@ -11,7 +11,9 @@ module AutoLocale
     private
 
     def set_locale
-      I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
+      hal = http_accept_language
+      hal.header ||= I18n.default_locale.to_s
+      I18n.locale = hal.compatible_language_from(I18n.available_locales)
```

Comment:
@DouweM Thanks for taking a look!  I'm afraid it wouldn't, because `compatible_language_from` may return nil when the Accept-Language header is given and contains no available locales.  Applying a fallback to those cases as well will spoil the `enforce_available_locales` concept.